### ISSUE TITLE
Support multiple and divide on intervals

### ISSUE THREAD

**Issue by samuelcolvin** [apache/arrow-rs#6335]
In Postgres I can do:

```sql
select 3 * interval '1 day';
select interval '1 day' * 3;  -- same
select 1.23 * interval '1 day';
-- and divide
select interval '1 day' / 3;
```

I started trying to add support for this to datafusion but discovered that the operations aren't supported in arrow-rs and adding them (with support for all numeric types) is far from trivial.

**Comment by samuelcolvin** [comment]
I guess we don't need all numeric types, just i64 and f64 scalars would go a long way.

**Comment by alamb** [comment]
> I guess we don't need all numeric types, just i64 and f64 scalars would go a long way.

Yes I think supporting divide for those types would be great (and then datafusion can cast other arguments to one of them in the coercion logic)

**Comment by milevin** [comment]
take

**Comment by peterxcli** [comment]
Hi, I opened a PR for interval multiplication with integer: https://github.com/apache/arrow-rs/pull/10336, which I think would be easy to review and no time unit issue that was previously discussed in https://github.com/apache/arrow-rs/pull/6906.
